src/WebJobs.Script.WebHost/Diagnostics/ConsoleWriter.cs

```diff
@@ -0,0 +1,189 @@
+                    var s when int.TryParse(s, out int i) && i > 0 => Channel.CreateBounded<string>(i),
+                    _ => Channel.CreateBounded<string>(DefaultBufferSize),  // default behavior is to use buffer with default size
+                };
+
+                if (_consoleBuffer == null)
+                {
+                    _writeEvent = Console.WriteLine;
+                }
+                else
+                {
+                    _writeEvent = WriteToConsoleBuffer;
+                    _consoleBufferTimeout = consoleBufferTimeout;
+                    if (autoStart)
+                    {
+                        bool batched = environment.GetEnvironmentVariable(EnvironmentSettingNames.ConsoleLoggingBufferBatched) switch
+                        {
+                            "0" => false,     // disable batching by setting to 0
+                            _ => true,        // default behavior is batched
+                        };
+
+                        StartProcessingBuffer(batched);
+                    }
+                }
+            }
+            else
+            {
+                _writeEvent = (string s) => { };
+            }
+
+            _exceptionhandler = exceptionHandler;
+        }
+
+        public void WriteHandler(string evt)
+        {
+            _writeEvent(evt);
+        }
+
+        private void WriteToConsoleBuffer(string evt)
+        {
+            try
+            {
+                while (_consoleBuffer.Writer.TryWrite(evt) == false)
+                {
+                    // Buffer is currently full, wait until writing is permitted.
+                    // This is the downside of using channels, we are on a sync code path and so we have to block on this task
+                    var writeTask = _consoleBuffer.Writer.WaitToWriteAsync().AsTask();
+                    if (writeTask.WaitAsync(_consoleBufferTimeout).Result == false)
+                    {
+                        // The buffer has been completed and does not allow further writes.
+                        // Currently this should not be possible, but the safest thing to do is just write directly to the console.
+                        Console.WriteLine(evt);
+                        break;
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // Most likely a task cancellation exception from the timeout expiring, but regardless we use the handler
+                // dump the raw exception and write the event to the console directly
+                _exceptionhandler(ex);
+                Console.WriteLine(evt);
+            }
+        }
+
+        internal void StartProcessingBuffer(bool batched)
+        {
+            // intentional no-op if the task is already running
+            if (_consoleBufferReadLoop == null || _consoleBufferReadLoop.IsCompleted)
```

Comment:
Alternative way via pattern matching:

```suggestion
            if (_consoleBufferReadLoop is { IsCompleted: true })
```

Reply:
This is an OR, while I think your suggestion is for an AND?

Reply:
Ah yes, I missed that.